watertap/unit_models/tests/test_coag_floc_model.py

```diff
@@ -9,585 +9,167 @@
+    m.fs.unit.inlet.flow_mass_phase_comp[0, "Liq", "TDS"].fix(0.01)
+    m.fs.unit.inlet.flow_mass_phase_comp[0, "Liq", "TSS"].fix(tss_in)
+    m.fs.unit.inlet.flow_mass_phase_comp[0, "Liq", "Sludge"].fix(0.0)
+
+    # set performance vars
+    m.fs.unit.rapid_mixing_retention_time[0].fix(60)
+    m.fs.unit.num_rapid_mixing_basins.fix(4)
+    m.fs.unit.rapid_mixing_vel_grad[0].fix(750)
+
+    m.fs.unit.floc_retention_time[0].fix(1800)
+    m.fs.unit.single_paddle_length.fix(4)
+    m.fs.unit.single_paddle_width.fix(0.5)
+    m.fs.unit.paddle_rotational_speed[0].fix(0.03)
+
+    m.fs.unit.paddle_drag_coef[0].fix(1.5)
+    m.fs.unit.vel_fraction.fix(0.7)
+    m.fs.unit.num_paddle_wheels.fix(4)
+    m.fs.unit.num_paddles_per_wheel.fix(4)
+
+    iscale.set_scaling_factor(
+        m.fs.unit.control_volume.properties_in[0.0].mass_frac_phase_comp[
+            "Liq", "Sludge"
+        ],
+        1e12,
+    )
+    iscale.calculate_scaling_factors(m.fs.unit)
+
+    return m
+
+
+class TestCoagFloc(UnitTestHarness):
+    def configure(self):
+        m = build()
+
+        self.unit_solutions[m.fs.unit.outlet.flow_mass_phase_comp[0, "Liq", "H2O"]] = 1
+        self.unit_solutions[
+            m.fs.unit.outlet.flow_mass_phase_comp[0, "Liq", "Sludge"]
+        ] = 0.009990636
+        self.unit_solutions[m.fs.unit.outlet.flow_mass_phase_comp[0, "Liq", "TDS"]] = (
+            0.01001510
+        )
+        self.unit_solutions[m.fs.unit.outlet.flow_mass_phase_comp[0, "Liq", "TSS"]] = (
+            9.36352627e-6
+        )
 
-# -----------------------------------------------------------------------------
-# Start test class with bad config
-class TestCoagulation_withBadProperties:
-    @pytest.fixture(scope="class")
-    def coag_obj_bad_properties(self):
-        model = ConcreteModel()
-        model.fs = FlowsheetBlock(dynamic=False)
+        return m
 
-        return model
 
-    @pytest.mark.unit
-    def test_build_model_catch_prop_errors(self, coag_obj_bad_properties):
-        model = coag_obj_bad_properties
+class TestCoagFlocNoChemicals(UnitTestHarness):
+    def configure(self):
+        m = build_no_chemicals()
 
-        error_msg = (
-            "Coagulation-Flocculation model MUST contain ('Liq','TDS') "
-            "as a component, but the property package has only specified "
-            "the following components [('Liq', 'H2O'), ('Liq', 'NaCl')]"
+        self.unit_solutions[m.fs.unit.outlet.flow_mass_phase_comp[0, "Liq", "H2O"]] = 1
+        self.unit_solutions[
+            m.fs.unit.outlet.flow_mass_phase_comp[0, "Liq", "Sludge"]
+        ] = 0.009112749
+        self.unit_solutions[m.fs.unit.outlet.flow_mass_phase_comp[0, "Liq", "TDS"]] = (
+            0.01
         )
-        with pytest.raises(ConfigurationError, match=re.escape(error_msg)):
-            model.fs.properties = NaClParameterBlock()
-            model.fs.unit = CoagulationFlocculation(
-                property_package=model.fs.properties
-            )
-
```

Comment:
Where are we maintaining the checks on ConfigurationError and other expected exception handling? Is that part of the test harness?

Reply:
Yea, my comment was also going to be to re-add this in. [I just appended mine at the end of GAC after the harness](https://github.com/watertap-org/watertap/blob/main/watertap/unit_models/tests/test_gac.py#L312). Thoughts if that is the best implementation?

Reply:
Seems like a good implementation to me. I've addressed this in the latest commit.